### ISSUE TITLE
指标相差太大是什么原因呢？

### ISSUE THREAD

**Issue by luyao-cv** [Yuliang-Liu/Monkey#129]
![image](https://github.com/user-attachments/assets/46582476-636f-4375-921e-e7e402ee707a)

你好，我复现评测指标，用的MultimodalOCR工具，测出来的点（第二行）跟你们论文的指标相差很大。组网代码用的你们的chat_internl

STVQAacc分数 | POIEacc分数 | FUNSDacc分数 | SROIEacc分数
-- | -- | -- | --
  | 69.9 | 42.9 | 70.3
64.2 | 47.8 | 40.6 | 63.4

**Comment by mxin262** [comment]
Hi~, we upload the evaluation file in the [MultimodalOCR](https://github.com/Yuliang-Liu/MultimodalOCR/blob/main/scripts/MiniMonkey.py)

**Comment by luyao-cv** [comment]
好的， 我试下，另外，这些指标仍然对不齐是为什么呢？第一行是您论文的数据，第二行是我用你们代码测出来的。

test: 87.4 | 76.5 | test: 60.1 | 75.7 | 802
-- | -- | -- | -- | --
86.9 | 74.3 | 57.1 | 73.5 | 800

**Comment by luyao-cv** [comment]
> Hi~, we upload the evaluation file in the [MultimodalOCR](https://github.com/Yuliang-Liu/MultimodalOCR/blob/main/scripts/MiniMonkey.py)

您好，使用了官方提供的权重，
用的 [MultimodalOCR](https://github.com/Yuliang-Liu/MultimodalOCR/blob/main/scripts/MiniMonkey.py) 测试，python ./scripts/MiniMonkey.py --image_folder ./OCRBench_Images/data --OCRBench_file ./OCRBench/FullTest.json  --num_workers 1 测出来的FUNSD: 0.419953596287703，与论文指标42.9有差异，请问是用的什么环境呢。
cuda 11.8
python 3.10
transformers 4.39.3
flash-atten: 3.6.3 (编译的)

**Comment by mxin262** [comment]
We use
flash-attn   2.5.8
pytorch      2.2.2
transformers  4.40.1
cuda  11.8
python    3.10.14

The FUNSD can achieve  0.43155452436194897.

For VLMEvalKit, you can replace the  VLMEvalKit/vlmeval/vlm/internvl_chat.py with this  [file](https://huggingface.co/mx262/MiniMokney/blob/main/internvl_chat.py). It achieves 806 on OCRBench, 76.0 on TextVQA_VAL, and 60.2 on InfoVQA_VAL

For the docvqa-test and chartqa, we evaluate them following this [step](https://internvl.readthedocs.io/en/latest/internvl2.0/evaluation.html#evaluation-using-internvl-codebase)

**Comment by luyao-cv** [comment]
请问当数据集为ChartQA_TEST的时候，self.min_num和self.min_num2要设置为多少，才能测出76的精度呢？

**Comment by luyao-cv** [comment]
关于docvqa-test 和 chartqa，我尝试了在internvl2代码库的eval代码里加上dynamic_preprocess2处理，使用https://www.wisemodel.cn/models/HUST-VLRLab/Mini-Monkey/file 权重，最高docvqa-test交官网也只有86.96(论文里87.4)， chartqa最高只有76.2(论文里76.5)，希望您可以公布下可以准确复现这两个精度的代码和参数，谢谢

**Comment by mxin262** [comment]
Hi~, we update the evaluation  [file](https://github.com/Yuliang-Liu/MultimodalOCR/blob/main/scripts/MiniMonkey.py) [](https://github.com/Yuliang-Liu/Monkey/blob/main/project/mini_monkey/eval/vqa/evaluate_vqa.py)

**Comment by LiWentomng** [comment]
@luyao-cv hello,  I re-retrained  mini-monkey followining the provided details, and use VLMEvalKit to perform evaluation. 
The results are listed below:

|  | MME |  SEED-IMG |Textvqa-val |OCR-Bench|
| :-----| ----: | :----: |:----: |:----: |
| paper | 1881.9 | 71.3 |75.7|802|
| re-train | 1842| 71.1 |71.9|632|

`paper` denotes the reported results  from paper, `re-train `is my testing results based on retraining model.

Can you provide some suggestions? Thanks.

**Comment by mxin262** [comment]
Hi~, can you provide the training log? @LiWentomng

**Comment by LiWentomng** [comment]
@mxin262  this is my training log. thanks for your time.
[training_log.txt](https://github.com/user-attachments/files/16823830/training_log.txt)

**Comment by mxin262** [comment]
@LiWentomng Hello, you might need to adjust the learning rate to 4e-9. I was able to reproduce the results using this learning rate on 4 GPUs.

**Comment by luyao-cv** [comment]
非常感谢提供的帮助，我这边再试试。
